R/req-url.R

```diff
@@ -67,14 +64,20 @@ req_url_path <- function(req, ...) {
 req_url_path_append <- function(req, ...) {
   check_request(req)
   path <- paste(..., sep = "/")
+  path <- path_add_slash(path)
 
   url <- url_parse(req$url)
 
+  url$path <- paste0(sub("/$", "", url$path), path)
+
+  req_url(req, url_build(url))
+}
+
+path_add_slash <- function(path) {
```

Comment:
I think we should make this `path_add_leading_slash()` just to be very specific.